dvc/api.py

```diff
@@ -69,3 +72,28 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, repo=None, rev=None):
+    # 1. Read grimoire.yaml
+    # 2. Pull dependencies (TODO)
+    # 3. Get the call and parameters
+    # 4. Invoke the call with the given parameters
+    # 5. Return the result
+
```

Comment:
Also, we need a way to overwrite params.

For example, `Dvc.summon` contains `params: {pretrained: True, num_classes: 1000}` (see in `dvc-pretrained-resnet` repo). User should be able to overwrite the params: `summon('resnet50', my_repo, params={'num_classes': 59}):`